commit d5e2df9ff5079b52d22a98bb74b95bb0df602b65
Author: Jian Shen <shenjian15@huawei.com>
Date:   Thu Dec 20 11:52:01 2018 +0800

    net: hns3: add max vector number check for pf
    
    [ Upstream commit 75edb610860fda65ceedb017fc69afabd2806b8b ]
    
    Each pf supports max 64 vectors and 128 tqps. For 2p/4p core scenario,
    there may be more than 64 cpus online. So the result of min_t(u16,
    num_Online_cpus(), tqp_num) may be more than 64. This patch adds check
    for the vector number.
    
    Fixes: dd38c72604dc ("net: hns3: fix for coalesce configuration lost during reset")
    Signed-off-by: Jian Shen <shenjian15@huawei.com>
    Signed-off-by: Peng Li <lipeng321@huawei.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/drivers/net/ethernet/hisilicon/hns3/hns3_enet.c b/drivers/net/ethernet/hisilicon/hns3/hns3_enet.c
index d5f15d8d18a9..0ccfa6a84535 100644
--- a/drivers/net/ethernet/hisilicon/hns3/hns3_enet.c
+++ b/drivers/net/ethernet/hisilicon/hns3/hns3_enet.c
@@ -2691,6 +2691,8 @@ static int hns3_nic_init_vector_data(struct hns3_nic_priv *priv)
 
 static int hns3_nic_alloc_vector_data(struct hns3_nic_priv *priv)
 {
+#define HNS3_VECTOR_PF_MAX_NUM		64
+
 	struct hnae3_handle *h = priv->ae_handle;
 	struct hns3_enet_tqp_vector *tqp_vector;
 	struct hnae3_vector_info *vector;
@@ -2703,6 +2705,8 @@ static int hns3_nic_alloc_vector_data(struct hns3_nic_priv *priv)
 	/* RSS size, cpu online and vector_num should be the same */
 	/* Should consider 2p/4p later */
 	vector_num = min_t(u16, num_online_cpus(), tqp_num);
+	vector_num = min_t(u16, vector_num, HNS3_VECTOR_PF_MAX_NUM);
+
 	vector = devm_kcalloc(&pdev->dev, vector_num, sizeof(*vector),
 			      GFP_KERNEL);
 	if (!vector)
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1177_d5e2df9ff5079b52d22a98bb74b95bb0df602b65.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1177_d5e2df9ff5079b52d22a98bb74b95bb0df602b65.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
